interoperator/pkg/internal/config/config.go

```diff
@@ -98,16 +118,102 @@ func (cfg *config) GetConfig() *InteroperatorConfig {
 			continue
 		}
 		f := reflect.Indirect(val).FieldByName(field.Name)
-		if f.Kind() == reflect.Int {
-			intVal, err := strconv.Atoi(fieldVal)
-			if err != nil {
-				log.Error(err, "invalid config value, skipping", "field", key, "value", fieldVal)
-				continue
-			}
-			if f.CanSet() {
-				f.SetInt(int64(intVal))
-			}
-		}
+		decodeField(f, fieldVal)
 	}
 	return interoperatorConfig
 }
+
+func (cfg *config) UpdateConfig(interoperatorConfig *InteroperatorConfig) error {
+	if interoperatorConfig == nil {
+		return errors.NewInputError("UpdateConfig", "interoperatorConfig", nil)
+	}
+	err := cfg.fetchConfig()
+	if err != nil && !apiErrors.IsNotFound(err) {
+		log.Error(err, "failed to fetch interoperator config for update")
+		return err
+	}
+
+	toCreate := false
+	if apiErrors.IsNotFound(err) {
+		toCreate = true
+		cfg.configMap = &corev1.ConfigMap{}
+		cfg.configMap.SetName(constants.ConfigMapName)
+		cfg.configMap.SetNamespace(cfg.namespace)
+		cfg.configMap.Data = make(map[string]string)
+	}
+
+	val := reflect.ValueOf(interoperatorConfig)
```

Comment:
If we dump the whole yaml as a single key/value in the `ConfigMap` as mentioned [above](https://github.com/cloudfoundry-incubator/service-fabrik-broker/pull/622/files#diff-5c71366ff7ef7d1a65501881ecb6544cR7), we can then simply use YAML/JSON (en|de)coder on the whole thing instead the current mixture of `reflect` and (en|de)coding.

Reply:
Use of reflect removed